video/out/opengl/egl_helpers.c

```diff
@@ -78,6 +78,20 @@ static void dump_egl_config(struct mp_log *log, int msgl, EGLDisplay display,
     }
 }
 
+static void *mpegl_get_proc_address(void *ctx, const char *name)
+{
+    void *p = eglGetProcAddress(name);
+#if defined(__GLIBC__) && HAVE_LIBDL
+    // Some crappy ARM/Linux things do not provide EGL 1.5, so above call does
+    // not necessarily return function pointers for core functions. Try to get
+    // them from a loaded GLES lib. As POSIX leaves RTLD_DEFAULT "reserved",
+    // use it only with glibc.
+    if (!p)
+        p = dlsym(RTLD_DEFAULT, name);
+#endif
```

Comment:
Since you're moving this, might as well make it more useful.
 - the rant doesn't help much document what's happening - see an example from another project below
 - kill off the unused ctx argument
```
   - Mali drivers on EGL 1.4 expose GL API statically from
     libGLESv2 but not dynamically from eglGetProcAddress. The EGL 1.4 spec
     permits this behavior.
    - EGL 1.5 requires that eglGetStringi be exposed dynamically through
      eglGetProcAddress. Exposing statically with dlsym is optional.
   - Mesa drivers expose GL API statically from libGL and libGLESv2
     and dynamically from eglGetProcAddress and glxGetProcAddress.
```

Reply:
I didn't originally write this. ;) I think the void *ctx is used with the function pointer stuff in `opengl/common.c`, so it can't be removed.

Reply:
Yup, perfectly aware you didn't write the original. Looking at the `opengl/common.c` code - seems like the most convoluted GL handling out there. As said originally it's a suggestion - feel free to take it or leave it. 
 https://github.com/mpv-player/mpv/blob/8a5e3d194b25471b540bfaca7c4c2b1707ba37ae/video/out/opengl/common.c#L680